src/ui.rs

```diff
@@ -1,10 +1,93 @@
-use bevy::prelude::*;
-
 use std::collections::HashMap;
 
+use bevy::input::mouse::{MouseMotion, MouseWheel};
+use bevy::prelude::*;
+
 use crate::constants::{get_element_color, get_element_size};
 use crate::structure::{AtomEntity, Crystal};
 
+/// Identifier for a reusable toggle interaction.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum ToggleId {
+    LightAttachment,
+    CameraControls,
+}
+
+impl ToggleId {
+    fn label(self, state: bool) -> &'static str {
+        match (self, state) {
+            (ToggleId::LightAttachment, true) => "Light: Attached",
+            (ToggleId::LightAttachment, false) => "Light: Detached",
+            (ToggleId::CameraControls, true) => "Controls: Keyboard",
+            (ToggleId::CameraControls, false) => "Controls: Mouse",
+        }
+    }
+}
+
+/// Stores the current on/off state for each toggle.
+#[derive(Resource, Default)]
+pub struct ToggleStates {
```

Comment:
```suggestion
pub(crate) struct ToggleStates {
```

can you try to go through all changes and see if `pub(crate)` is enough and can event make it private API?